packages/hydrogen/src/storefront.ts

```diff
@@ -42,17 +42,76 @@ export type StorefrontClient<TI18n extends I18nBase> = {
   storefront: Storefront<TI18n>;
 };
 
+// Default type for `variables` in storefront client
+type GqlVariables = ExecutionArgs['variableValues'];
+
+// Use this type to make parameters optional in storefront client
+// when no variables need to be passed.
+type EmptyVariables = {[key: string]: never};
+
+// These are the variables that are automatically added to the storefront API.
+// We use this type to make parameters optional in storefront client
+// when these are the only variables that can be passed.
+type AutoAddedVariableNames = 'country' | 'language';
+
+// This interface will be augmented in user land with generated query types
+export interface QueryTypes<U = any> {
+  [key: string]: {
+    return: U;
+    variables: {[key: string]: any};
+    // This property *is not* added by the codegen plugin so it can be used
+    // to discriminate between this default type and generated types.
+    forceOptionalVariables: true;
```

Comment:
This one in particular is just a private option, it should never appear in user-land. We just use it to determine if the current type is the default one (the generic type included in @shopify/hydrogen for every query) or a generated one -- generated types in user-land don't include this option, so it will be `undefined`.